docs/source/faq.rst

```diff
@@ -39,6 +39,52 @@ As a consequence, the following code is correct too:
 More details `Events and Handlers <concepts.html#events-and-handlers>`_.
 
 
+Creating Custom Events based on Forward/Backward Pass
+-----------------------------------------------------
+
+There are cases where the user might want to add events based on the loss calculation and backward pass. Ignite provides
+flexibility to the user to allow for this:
+
+.. code-block:: python
+
+    class BackpropogationEvents(Enum):
+        """
+        Events based on Backpropogation
+        """
+        BACKPROPOGATION_STARTED = 'backpropogation_started'
+        BACKPROPOGATION_ENDED = 'backpropogation_ended'
+
+
+    def update(engine, batch):
+        model.train()
+        opitmizer.zero_grad()
+        x, y = process_batch(batch)
+        y_pred = model(x)
+        loss = loss_fn(y_pred, y)
+        engine.fire_event(BackpropogationEvents.BACKPROPOGATION_STARTED)
+        loss.backward()
+        optimizer.step()
```

Comment:
https://docs.fast.ai/callback.html#Callback.on_backward_end
engine.fire_event(BackpropogationEvents.BACKPROPOGATION_ENDED) should be fired before optimizer.step()